src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -3187,5 +3187,7 @@ public static string GetRequiredLanguageVersion(Diagnostic diagnostic)
 
             return foundVersion;
         }
+
+        public abstract INamespaceDeclaration DeclarationRoot { get; }
```

Comment:
>INamespaceDeclaration [](start = 24, length = 21)

is this necessarily a namespace, or could it be a type too? (top-level type scenario)

Reply:
it shoudl always be a namespace since it's the root-namespace (even if you have a top-level type).

Reply:
We should also consider exposing:

`IDeclaration.NameLocations` and `ITypeDeclaration.SyntaxReferences`

It seems unfortunate that you could enumerate these guys, but not actually ever get to them!

Reply:
That occurred to me, but I'm OK with leaving them out until a need arises.

---
In reply to: [267135347](https://github.com/dotnet/roslyn/pull/34067#discussion_r267135347) [](ancestors = 267135347)